backend/src/main/java/de/bund/digitalservice/ris/norms/domain/entity/CharacterRange.java

```diff
@@ -54,6 +59,121 @@ public boolean isValidCharacterRange() {
     return pattern.matcher(characterRange()).matches() && isEndGreaterStart();
   }
 
+  /**
+   * Finds the text-content of the range within the given node.
+   *
+   * @param node the node in which the range should be found
+   * @return the text-content of the range
+   */
+  public String findTextInNode(Node node) {
+    var text =
+        getNodesInRange(node).stream().map(Node::getTextContent).collect(Collectors.joining());
+    node.normalize();
+    return text;
+  }
+
+  /**
+   * Splits the given text node into multiple text nodes and returns the text node that includes the
+   * range from start to end of the text content.
+   *
+   * <p>A call to Node::normalize undoes this.
+   *
+   * @param textNode A text node
+   * @param start start of the text range within the node
+   * @param end end (excluding) of the text range within the node
+   * @return a text node that represents exactly the range
+   */
+  private Node extractTextNodeForRange(Node textNode, int start, int end) {
+    final var textContent = textNode.getTextContent();
+
+    // text before selection
+    if (start != 0) {
+      textNode
+          .getParentNode()
+          .insertBefore(
+              textNode.getOwnerDocument().createTextNode(textContent.substring(0, start)),
+              textNode);
+    }
+
+    // selected text
+    final var rangeTextContent = textContent.substring(start, Math.min(textContent.length(), end));
+    final var rangeNode =
+        textNode
+            .getParentNode()
+            .insertBefore(textNode.getOwnerDocument().createTextNode(rangeTextContent), textNode);
+
+    // text after selection
+    if (end < textContent.length()) {
+      textNode
+          .getParentNode()
+          .insertBefore(
+              textNode.getOwnerDocument().createTextNode(textContent.substring(end)), textNode);
+    }
+
+    textNode.getParentNode().removeChild(textNode);
+
+    return rangeNode;
+  }
+
+  /**
+   * Finds the nodes that make up the given range within the given node.
+   *
+   * <p>SIDE-EFFECT: This might split text-nodes into multiple text nodes to be able to return
+   * exactly the range. Please call Node::normalize after you are finished working with the nodes.
+   *
+   * @param node the node in which the range should be found
+   * @return a list of nodes that represent the given range
+   */
+  public List<Node> getNodesInRange(Node node) {
+    if (node.getTextContent().length() < getStart()) {
+      throw new IndexOutOfBoundsException();
```

Comment:
It would be nice if we would use the other constructor of that exception to pass a message giving specific information like what was the length of the node and what was the given start?